File: client/pulse/lib/model/post.dart

import 'package:pulse/model/user.dart';

class Post {

  final String postId;
  final User user;
  final String description;
  final String picturePath;

  Post(this.postId, this.user, this.description, this.picturePath);

  factory Post.fromJson(Map<String, dynamic> json, String token) {
    return Post(
        json["id"],
        User.fromToken({"token": token}),
        json["postDescription"],
        json["picturePath"]
    );
  }

}

File: server/pulse/src/main/java/com/bant/pulse/model/post/Post.java
package com.bant.pulse.model.post;

import com.bant.pulse.model.likes.Like;
import com.bant.pulse.model.user.User;
import jakarta.persistence.*;
import jakarta.transaction.Transactional;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;
import java.util.UUID;

@Builder
@AllArgsConstructor
@NoArgsConstructor
@Data
@Entity
@Transactional
@Table(
        name = "pulse_posts"
)
public class Post {

    @Id
    @GeneratedValue(
            strategy = GenerationType.UUID
    )
    private UUID id;

    @ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn(
            name = "user_id",
            nullable = false
    )
    private User user;

    private String postDescription;
    private String picturePath;

    @OneToMany(cascade = CascadeType.ALL)
    @JoinColumn(
            name = "post_likes",
            referencedColumnName = "id"
    )
    private List<Like> likes;
}
